src/osd/PrimaryLogPG.cc

```diff
@@ -2371,6 +2371,11 @@ PrimaryLogPG::cache_result_t PrimaryLogPG::maybe_handle_manifest_detail(
     }
   }
 
+  if (!obc || !obc->obs.oi.has_manifest()) {
```

Comment:
How good it would be place this check at start of `PrimaryLogPG::maybe_handle_manifest_detail`